src/test/perl/test-caffileeditor_source.t

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use FindBin qw($Bin);
+use lib "$Bin/modules";
+use CAF::FileEditor;
+use Test::More tests => 7;
+use Test::MockModule;
+use Test::Quattor::Object;
+use Carp qw(confess);
+use File::Path;
+use File::Temp qw(tempfile);
+use Readonly;
+
+$SIG{__DIE__} = \&confess;
+
+# FIXME:
+# LC::File::file_contents (used by CAF::FileEditor constructor) doesn't work 
+# in the unit test context. Mock it until we find a better solution...
+our $lcfile = Test::MockModule->new("LC::File");
+
+sub read_file_contents {
+    my $fname = shift;
+    my $fh;
+    open($fh, "<", $fname) || die ("failed to open $fname");
+    my $contents = join('', <$fh>);
+    return $contents;
+}
+$lcfile->mock("file_contents", \&read_file_contents);
+
+
+my $testdir = 'target/test/editor';
+mkpath($testdir);
+(undef, my $filename) = tempfile(DIR => $testdir);
+
+Readonly my $TEXT => <<EOF;
+En un lugar de La Mancha, de cuyo nombre no quiero acordarme
+no ha tiempo que vivía un hidalgo de los de lanza en astillero...
+EOF
+Readonly my $ANOTHER_TEXT => "adarga antigua, rocín flaco y galgo corredor.";
+
+my $fh;
+my $obj = Test::Quattor::Object->new();
+
+
+# Create a file and check that it is empty
+($fh, $filename) = tempfile(DIR => $testdir);
+$fh->close();
+$fh = CAF::FileEditor->new($filename, log => $obj);
+is("$fh","","Existing file ($filename) empty");
+$fh->close();
+
+# Check that reference file contents is used as the initial contents when it
+# is newer than the file edited.
+sleep 2;
```

Comment:
it adds 4 seconds to the tests, and setting the mtime by hand to some non-sensical value is much better way to control it than some random sleep value. it's as simple as
```perl
$time=time()
utime(undef, $time - 10, $filename); # make filename old enough
```